db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DeleteNode.java

```diff
@@ -964,12 +974,7 @@ private StatementNode getEmptyUpdateNode(String schemaName,
                                                      null, /* windows */
                                                      getContextManager());
 
-        return (StatementNode) nodeFactory.getNode(
-                                                    C_NodeTypes.UPDATE_NODE,
-                                                    tableName,
-                                                    resultSet,
-                                                    getContextManager());
-
+        return new UpdateNode( tableName, resultSet, cursorDelete, getContextManager());
```

Comment:
this is actually fixing a bug that was introduced because NodeFactory does no type checking (UpdateNode's init requires `tableName, resultSet, cursorDelete`, but would have only been given `tableName, resultSet` )